pkg/proxy/service.go

```diff
@@ -43,12 +141,20 @@ type ServiceChangeTracker struct {
 	lock sync.Mutex
 	// items maps a service to its serviceChange.
 	items map[types.NamespacedName]*serviceChange
+	// newServiceInfo allows proxier to inject customized information when processing service.
+	newServiceInfo newServiceInfoFunc
```

Comment:
newServiceInfoFunc still outputs proxy.ServicePort. 

So I would say `makeServicePort` is more appropriate.

Reply:
Good point, changing back to `makeServicePortFunc`.

Reply:
Done.